util/stopwaiter/stopwaiter_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package stopwaiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util/testhelpers"
+)
+
+const testStopDelayWarningTimeout = 350 * time.Millisecond
+
+type TestStruct struct{}
+
+func TestStopWaiterStopAndWaitTimeoutShouldWarn(t *testing.T) {
+	logHandler := initTestLog(t, log.LvlTrace)
+	sw := StopWaiter{}
+	sw.Start(context.Background(), &TestStruct{})
+	sw.LaunchThread(func(ctx context.Context) {
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
to avoid race condition altogether- you don't need to check ctx at all.
Add a separate context that will cause this thread to exit after the test ends.

Reply:
this thread can be as simple as
```
{
 <-testctx.Done()
}
```
With testctx being a contextx created for this test and aborted when it ends (using context.WithCancel)

Reply:
I rewrote this part of the test, trying to apply the proposal